src/pipelinedefinition.hpp

```diff
@@ -70,6 +74,11 @@ class PipelineDefinition {
     std::vector<NodeInfo> nodeInfos;
     pipeline_connections_t connections;
 
+protected:
+    tensor_map_t inputsInfo;
+    tensor_map_t outputsInfo;
```

Comment:
What happens on reload? Is it possible that inputs / outputs change in the runtime. Are those fields updated then?

Reply:
Those are updated during PipelineDefinition.validate(). This is called each time that any change in config related to DAG is changed.